openmc/deplete/microxs.py

```diff
@@ -0,0 +1,229 @@
+"""MicroXS module
+
+A pandas.DataFrame storing microscopic cross section data with
+nuclides names as row indices and reaction names as column indices.
+"""
+
+import tempfile
+from pathlib import Path
+
+from pandas import DataFrame, read_csv, concat
+import numpy as np
+
+from openmc.checkvalue import check_type, check_value, check_iterable_type
+from openmc.mgxs import EnergyGroups, ArbitraryXS, FissionXS
+from openmc.data import DataLibrary
+from openmc import Tallies, StatePoint, Materials
+
+
+from .chain import Chain, REACTIONS
+from .coupled_operator import _find_cross_sections
+
+_valid_rxns = list(REACTIONS)
+_valid_rxns.append('fission')
+
+
+class MicroXS(DataFrame):
+    """Stores microscopic cross section data for use in
+    independent depletion.
+    """
+
+    @classmethod
+    def from_model(cls,
+                   model,
+                   reaction_domain,
+                   chain_file,
+                   dilute_initial=1.0e3,
+                   energy_bounds=(0, 20e6)):
+        """Generate a one-group cross-section dataframe using
+        OpenMC. Note that the ``openmc`` executable must be compiled.
+
+        Parameters
+        ----------
+        model : openmc.Model
+            OpenMC model object. Must contain geometry, materials, and settings.
+        reaction_domain : openmc.Material or openmc.Cell or openmc.Universe or openmc.RegularMesh
+            Domain in which to tally reaction rates.
+        chain_file : str
+            Path to the depletion chain XML file that will be used in depletion
+            simulation. Used to determine cross sections for materials not
+            present in the inital composition.
+        dilute_initial : float
+            Initial atom density [atoms/cm^3] to add for nuclides that
+            are zero in initial condition to ensure they exist in the cross
+            section data. Only done for nuclides with reaction rates.
+        reactions : list of str, optional
+            Reaction names to tally
+        energy_bound : 2-tuple of float, optional
+            Bounds for the energy group.
+
+        Returns
+        -------
+        MicroXS
+            Cross section data in [b]
+
+        """
+        groups = EnergyGroups(energy_bounds)
+
+        # Set up the reaction tallies
+        original_tallies = model.tallies
+        tallies = Tallies()
+        xs = {}
+        reactions, diluted_materials = cls._add_dilute_nuclides(chain_file,
+                                                                model,
+                                                                dilute_initial)
+        diluted_materials.export_to_xml('diluted_materials.xml')
```

Comment:
Doesn't look like this is used
```suggestion
```

Reply:
This was for testing purposed and forgot to remove it when I committed.